code/modules/unit_tests/closets.dm

```diff
@@ -0,0 +1,39 @@
+/*
+ * Unit test for checking that closets don't have more items in them
+ * when created, than they would be able to hold naturally.
+ * If a closet when created would have too many items, opening and closing
+ * it would leave some items on the floor.
+ */
+/datum/unit_test/closet_contents
+	// At time of writing, there are 185 unique closets. A 256 big
+	// area will be sufficient for now.
+	reservation_width = 18 // 2 is consumed by the border wall
+	reservation_height = 18
+	border_test_turf_type = /turf/closed/wall/r_wall
+	gravity = TRUE
+
+/datum/unit_test/closet_contents/Run()
+	var/list/test_turfs = block(run_loc_bottom_left, run_loc_top_right)
+
+	for(var/_closettype in typesof(/obj/structure/closet))
+		// Spawn each closet on its own turf, so they don't interfere.
+		var/turf/closet_turf
+		while(closet_turf == null || istype(closet_turf, /turf/closed))
```

Comment:
Right here, this is the infinite loop. I think it'd be better if you just had one turf, created the locker, opened/closed it, test, then qdel it.

Reply:
The problem I ran into, is that it assumes closets are "sensible" and "well behaved", and don't just randomly spawn other shit when created.

Reply:
Assuming the test doesn't infinite loop anymore, it feels to me that it would just break again once there are more than reversation^2 lockers/closets, which seems likely. 

I'd rather you just clean up everything on the tile in between closets.